src/Core/ServerSettings.h

```diff
@@ -103,6 +103,8 @@ namespace DB
     M(Bool, async_load_databases, false, "Enable asynchronous loading of databases and tables to speedup server startup. Queries to not yet loaded entity will be blocked until load is finished.", 0) \
     M(Bool, display_secrets_in_show_and_select, false, "Allow showing secrets in SHOW and SELECT queries via a format setting and a grant", 0) \
     \
+    M(Bool, force_read_through_cache_for_merges, false, "Force read-through filesystem cache for merges", 0) \
```

Comment:
I think it's better to make it `<merge_tree_setting>`.

Reply:
However it will be really hard implement....

Reply:
Not really, it is used in `MergeTreeSequentialSource`, which has `MergeTreeData` pointer already, which has the settings.